extension/persistence/relational-jdbc/src/main/java/org/apache/polaris/extension/persistence/relational/jdbc/ResultSetIterator.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.polaris.extension.persistence.relational.jdbc;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+import java.util.Spliterator;
+import java.util.Spliterators;
+import java.util.stream.Stream;
+import java.util.stream.StreamSupport;
+import org.apache.polaris.extension.persistence.relational.jdbc.models.Converter;
+
+/** Used to wrap a ResultSet and to build a stream from the data it contains */
+public class ResultSetIterator<T> implements Iterator<T> {
+  private final ResultSet resultSet;
+  private final Converter<T> converterInstance;
+  private boolean hasNext;
+
+  public ResultSetIterator(ResultSet resultSet, Converter<T> converterInstance) {
+    this.resultSet = resultSet;
+    this.converterInstance = converterInstance;
+    advance();
+  }
+
+  private void advance() {
+    try {
+      hasNext = resultSet.next();
+    } catch (SQLException e) {
+      hasNext = false;
+      throw new RuntimeException(e);
```

Comment:
interesting, if it's a SQLException shouldn't we throw it as SQLException to inspect the error code ? any other exception other than SQLException is considered as RTE for now in the caller.

Reply:
So the real problem here is `ResultSetIterator.next()`, which can't throw an exception so long as we're implementing `Iterator`. 

For now I'll have the constructor throw a `SQLException` so we have a chance of finding a `SQLException`, but we'd still wrap `SQLExceptions` that happen after the first `ResultSet.next()` call. If that's an issue, we can move away from implementing Iterator or fix the caller to unwrap the exception.